File: SecondYear/ApoorvaSri123_ApoorvaSrivastava_2226it1116_2/Week_2/armstrong.c
#include <stdio.h>
#include <math.h>

int main() {
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);

    int originalNum = num;
    int len = 0;
    while (originalNum > 0) {
        originalNum /= 10;
        len++;
    }

    int sum = 0;
    int temp = num;
    while (temp > 0) {
        int curr = temp % 10;
        sum += pow(curr, len);
        temp /= 10;
    }

    if (sum == num) {
        printf("true\n");
    } else {
        printf("false\n");
    }

    return 0;
}

File: SecondYear/Divyansh670_DivyanshSrivastav_2226it1029_2/Week_2/armstrong.java
import java.util.*;

 

public class armstrong {

    

    public static void main(String[] args) {

        try (Scanner sc = new Scanner(System.in)) {
            int num=sc.nextInt();

 

            String str=Integer.toString(num);

            int len=str.length();

            int sum=0;

 

            for(int i=0;i<len;i++){

                int curr=str.charAt(i)-'0';

                sum+=Math.pow(curr, len);

            }

 

            if(sum==num){

                System.out.println("true");

            }else{

                System.out.println("false");    

            }
        }

    }

}